### PR TITLE
Fix autorequire in puppet4

### PR DESCRIPTION
**Pull request by jyaworski**
Using puppet 4, I consistently get a failure to autoload with puppetserver. This fixes the problem.

### CODE REVIEW

**Comment by jyaworski** [comment]
Tag @liamjbennett and #146

**Comment by igalic** [comment]
@jyaworski could you do a rebase here? or is this already fixed through #146?

**Comment by jyaworski** [comment]
@igalic #146 doesn't fix it for me. I'd talked with @liamjbennett about alternatives. If I have time today I'll see if I can rebase this.

File: lib/puppet/parser/functions/validate_rd_policy.rb [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
-      Puppet::Util::RundeckACL.validate_acl(arg)
-    else
-      fail Puppet::ParseError, ("#{arg.inspect} is not a Hash or Array of hashes.  It looks to be a #{arg.class}")
+    fail Puppet::ParseError, ("validate_rd_policy(): wrong number of arguments (#{args.length}; must be 1)") unless args.length == 1
+
+    args.each do |arg|
+      next if arg.is_a?(Array)
+      if arg.is_a?(Hash)
+        Puppet::Util::RundeckACL.validate_acl(arg)
+      else
```
Comment (igalic) [inline lib/puppet/parser/functions/validate_rd_policy.rb]: fail early, remove a whole indentation level.

**Comment by igalic** [comment]
also: rebase again :O

**Comment by jyaworski** [comment]
Closing for now. I believe this is still a problem, but I don't have the facility to test any longer.

**Pull request closed without merging**